### PR TITLE
LSE beta doc updates

### PR DESCRIPTION
**Pull request by smoreface**
Making updates and fixing issues discovered with the latest iteration of the beta docs.
- Revise project publish info
- Revise review project and dashboard info
- Add details about project templates
- Update API docs to support LSE API improvements
- Add more details about securing access to cloud storage 
- Fixed broken links
- Made labeling workflow more obvious
- Add minimum Docker Compose version
- Update ML API docs to include programmatically assigning ML backend via API
- Add more details about assigning members to projects and tasks and workspaces
- Add additional LDAP environment variables for mapping roles
- Update install doc for LSE with more Docker info

### CODE REVIEW

File: label_studio/ml/api.py [hunk 1]
```diff
@@ -40,14 +41,24 @@ def perform_create(self, serializer):
...
+@method_decorator(name='perform_update', decorator=swagger_auto_schema(
+    tags=['Machine Learning'],
+    operation_summary='Add ML Backend',
+    operation_description="""
+    Add an ML backend using the Label Studio UI or by sending a POST request using the following cURL command:
+    ```bash
+    curl -X POST {host}/api/ml -H 'Authorization: Token abc123'\\
+    --data '{"url": "http://localhost:9090", "project": {{project_id}}}' 
+    """.format(host=(settings.HOSTNAME or 'https://localhost:8080')),
+))
```
Comment (smoreface) [inline label_studio/ml/api.py:53]: @makseq Please check this!

Comment (makseq) [reply label_studio/ml/api.py:53]: @smoreface looks like we can add a token as var too: 

```
curl -X POST {host}/api/ml -H 'Authorization: Token {token}'\\ .... .format(host=..., token='abc123')
```

Comment (smoreface) [reply label_studio/ml/api.py:53]: @makseq I agree, but I only want to do that if we can pull in the customer token dynamically like we're doing for the hostname. Is that stored somewhere we can access or not?

File: label_studio/tasks/api.py [hunk 2]
```diff
@@ -298,23 +298,29 @@ class AnnotationDraftAPI(RequestDebugLogMixin, generics.RetrieveUpdateDestroyAPI
...
-    tags=['Predictions'], operation_summary="Create prediction",
+    tags=['Predictions'],
+    operation_summary="Create prediction",
     operation_description="Create a prediction."))
 @method_decorator(name='retrieve', decorator=swagger_auto_schema(
-    tags=['Predictions'], operation_summary="Get prediction",
-    operation_description="Get all predictions."))
+    tags=['Predictions'],
+    operation_summary="Get prediction",
+    operation_description="Get all predictions for a specific task ID."))
```
Comment (smoreface) [inline label_studio/tasks/api.py:311]: @triklozoid can you clarify -- does this use the task ID, project ID, ML backend ID, or something else? It's unclear to me from the code for the serializer.

Comment (triklozoid) [reply label_studio/tasks/api.py:326]: Good question, now you can get only all predictions from organization. May be it's useful, I'll add filters by task and project.

https://github.com/heartexlabs/label-studio/pull/1130

How it works after this PR: 
```
GET /api/predictions  - all predictions from organization
GET /api/predictions?task=TASK_ID - all predictions for task with id TASK_ID
GET /api/predictions?task__project=PROJECT_ID - all predictions for project with id PROJECT_ID
```

Comment (smoreface) [reply label_studio/tasks/api.py:326]: Thank you!! I'll update accordingly :)

**Comment by smoreface** [comment]
@niklub / @makseq this PR is very big so I'm going to move any further updates to a new branch. Please review this one!!

**Comment by smoreface** [comment]
@niklub / @makseq Please make time to review this PR, there's lots of customer requests in here! I promise only small PRs from here on out 🥺

File: docs/source/guide/setup_project.md [hunk 3]
```diff
@@ -117,27 +106,32 @@ If you want, you can allow empty annotations.
...
+    3. Click **Save**. 
+    
 <i class='ent'></i> In Label Studio Enterprise, you can set up task sampling in the annotation settings for a project.
 1. Within a project on the Label Studio UI, click **Settings**.
 2. Click **Annotation Settings**.
 3. Select your preferred method of task sampling:
-- Uncertainty sampling, where tasks are shown to annotators according to the model uncertainty, or prediction scores.
-- Sequential sampling, the default, where tasks are shown to annotators in the same order that they appear on the Data Manager.
-- Uniform sampling, where tasks are shown to annotators in a random order.
+    - Uncertainty sampling, where tasks are shown to annotators according to the model uncertainty, or prediction scores.
```
Comment (niklub) [inline docs/source/guide/setup_project.md:120]: Uncertainty sampling is Enterprise only feature

File: docs/source/guide/setup_project.md [hunk 4]
```diff
@@ -165,12 +159,55 @@ Configure the reviewing settings for your project.
...
+    - Click **Import** to import data.
+    - Click **Data Manager** to start annotating.
+    
 
 ## Where Label Studio stores your project data and configurations
 
 All labeling activities in Label Studio occur in the context of a project.
 
-Starting in version 1.0.0, Label Studio stores your project data and configurations in a SQLite database. You can choose to use PostgreSQL or Redis instead. See [Setup database storage](storedata.html). 
+Starting in version 1.0.0, Label Studio stores your project data and configurations in a SQLite database. You can choose to use PostgreSQL or Redis instead. See [Set up database storage](storedata.html).
```
Comment (niklub) [inline docs/source/guide/setup_project.md:210]: you can't choose Redis as DB

File: docs/source/guide/storage.md [hunk 5]
```diff
@@ -15,18 +15,18 @@ Set up the following cloud and other storage systems with Label Studio:
 - [Redis database](#Redis-database)
 - [Local storage](#Local-storage)
 
-Each source and target storage setup is project-specific. You can connect multiple buckets, containers, databases, or directories as source or target storage for a project. 
+Each source and target storage setup is project-specific. You can connect multiple buckets, containers, databases, or directories as source or target storage for a project. If you upload new data to a connected cloud storage bucket, sync the storage connection to add the new labeling tasks to Label Studio without restarting. 
 
-If you upload new data to a connected cloud storage bucket, sync the storage connection to add the new labeling tasks to Label Studio without restarting. 
+> Choose your target storage carefully. When you start the labeling project, the target storage must be empty or contain annotations that match previously created or imported tasks from source storage. Tasks are synced with annotations based on internal IDs, so if you accidentally connect to target storage with existing annotations with the same IDs, the connection might fail with undefined behavior.
```
Comment (niklub) [inline docs/source/guide/storage.md:20]: syncing from Target Storage is deprecated since >=1.0, let's remove this block

**Review by niklub — CHANGES_REQUESTED** [review]

**Comment by niklub** [comment]
LGTM, let's fix small issues and finally push this

**Review by niklub — APPROVED** [review]

File: docs/source/guide/storage.md (hunk 5 above)
Comment (smoreface) [reply docs/source/guide/storage.md:20]: yes, I'm planning to do that in my next pass through of the storage doc

File: docs/source/guide/setup_project.md (hunk 3 above)
Comment (smoreface) [reply docs/source/guide/setup_project.md:120]: yes, this section has the enterprise badge on it.

File: docs/source/guide/setup_project.md (hunk 4 above)
Comment (smoreface) [reply docs/source/guide/setup_project.md:210]: Ooh yeah, will fix this.

**Pull request merged**